### PR TITLE
Add Docker image for backend

### PR DESCRIPTION
**Pull request by henrybrink**
As requested in #98 this PR adds a docker image for the backend, and a `docker-compose.yml` which can be used to start the application with postgres and a reverse proxy which both servers the frontend application and the backend application.

### CODE REVIEW

**Comment by henrybrink** [comment]
As Prisma for now refuses to work both with SQLite and Postgres, I remove the postgres from the docker-compose file for now.

File: docs/development/deployment.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Deployment
+
+Ziel ist es, die Anwendung während der Entwicklung auf einem Test-Server laufen zu lassen. Um die Aufgabe zu vereinfachen, soll die Anwendung mittels "Docker" gestartet werden können.
```
Comment (benedictweis) [inline docs/development/deployment.md:3]:
```suggestion
Ziel ist es, die Anwendung während der Entwicklung auf einem Test-Server laufen zu lassen. Um die Aufgabe zu vereinfachen, soll die Anwendung mittels [Docker](https://docs.docker.com/get-started/overview/) gestartet werden können.
```

😉

File: docs/development/deployment.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Um dies zu erreichen gibt es im Backend ein "Dockerfile", welches spezifische Build Anweisungen vorgibt. Dadurch wird ein Dockerimage für das Backend erstellt.
```
Comment (benedictweis) [inline docs/development/deployment.md:4]:
```suggestion
Um dies zu erreichen gibt es im Backend ein `Dockerfile`, welches spezifische Build Anweisungen vorgibt. Dadurch wird ein Dockerimage für das Backend erstellt.
```

File: docs/development/deployment.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+Mithilfe der "docker-compose.yml" im Wurzelverzeichnis der Anwendung kann das ganze Projekt gestartet werden.
```
Comment (benedictweis) [inline docs/development/deployment.md:5]:
```suggestion
Mithilfe der `docker-compose.yml` im Wurzelverzeichnis der Anwendung kann das ganze Projekt gestartet werden.
```

**Review by benedictweis — CHANGES_REQUESTED** [review]
Awesome PR, thanks! Just some minor nits.

File: docs/development/deployment.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+Nach dem starten der Container kann die Anwendung über `localhost:8000` aufgerufen werden. Die API wird über den Reverse-Proxy mit `/api/` bereitgestellt. Das Frontend direkt über `/`
```
[the remark this answers is not in the document]
Comment (henrybrink) [reply docs/development/deployment.md:7]: We already had the discussion about SSL for localhost :)

[the remark this answers is not in the document]
Comment (benedictweis) [reply docs/development/deployment.md]: Fine, changed to http

**Review by benedictweis — APPROVED** [review]

**Pull request merged**